src/model.d.ts

```diff
@@ -881,8 +898,37 @@ export interface FindOptions<TAttributes = any>
    */
   limit?: Nullish<number | Literal>;
 
-  // TODO: document this - this is an undocumented property but it exists and there are tests for it.
-  groupedLimit?: unknown;
+  /**
+   * groupedLimit allows you to do a limit for each key in groupedLimit.values, similar to what LATERAL JOIN / CROSS APPLY does. e.g:
+   *
+   * ```js
+   * this.projects = this.Project.findAll();
+   *
+   * this.User.findAll({
+   *    groupedLimit: {
+   *      limit: 3,
+   *      on: this.User.Projects,
+   *      values: this.projects.map(item => item.get('id'))
+   *    },
+   * });
+   * ```
+   * IN SQL
+   * ```sql
+   * SELECT * FROM (
+   *   UNION ALL
+   *    SELECT * FROM "user" WHERE project_id = 1 LIMIT 3,
+   *    SELECT * FROM "user" WHERE project_id = 2 LIMIT 3,
+   *    SELECT * FROM "user" WHERE project_id = 3 LIMIT 3
+   * );
+   * ```
+   *
+   * Since not every dialect supports lateral joins, this is currently implemented as a union.
```

Comment:
That's not the reason for the existence of `groupedLimit`, it's used internally when an association must be loaded in a separate query (`separate` option), you wouldn't do that with a lateral join (you'd use a separate join if `separate` were false)

```suggestion
```